src/app/pages/MediaAssetPage/MediaAssetPage.jsx

```diff
@@ -53,9 +53,18 @@ const MediaAssetPage = ({ pageData }) => {
     pageData,
   );
   const indexImagePath = path(['promo', 'indexImage', 'path'], pageData);
-  const indexImageLocator = indexImagePath
-    ? getImageParts(indexImagePath)[1]
-    : null;
+
+  const getIndexImageLocator = () => {
+    const imageParts = indexImagePath && getImageParts(indexImagePath);
+
+    if (isLegacyMediaAssetPage(requestContext.canonicalLink)) {
```

Comment:
Could bump this to line 58 so it gets hit first in the function & only does the getImageParts logic if it isn't a legacy asset? Marginal improvement 🤷‍♂️

Reply:
Good shout, will do that